src/Storefront/Controller/CartLineItemController.php

```diff
@@ -70,6 +70,14 @@ public function deleteLineItem(Cart $cart, string $id, Request $request, SalesCh
         });
     }
 
+    #[Route(path: '/checkout/cart/delete', name: 'frontend.checkout.cart.delete', defaults: ['XmlHttpRequest' => true], methods: ['POST'])]
```

Comment:
In `store-api` there is
`DELETE /checkout/cart`
I'd suggest to keep it that way also for the storefront endpoint for consistency
```suggestion
    #[Route(path: '/checkout/cart', name: 'frontend.checkout.cart.delete', defaults: ['XmlHttpRequest' => true], methods: ['DELETE'])]
```

Reply:
It's on purposes a POST, so you can build a regular HTML form on it

Reply:
why not allowing both? :thinking:

Reply:
thanks, for the storefront app that makes sense then 👍 

@mitelg you can already call store-api DELETE via JS client directly. Here we have dedicated storefront middleware for store-api so having a single opinionated way is cleaner for it if the pattern for forms is like this

Reply:
ah sure :man_facepalming: